fvm/evm/offchain/sync/replayer.go

```diff
@@ -45,22 +45,35 @@ func NewReplayer(
 }
 
 // ReplayBlock replays the execution of the transactions of an EVM block
+func (cr *Replayer) ReplayBlock(
+	transactionEvents []events.TransactionEventPayload,
+	blockEvent *events.BlockEventPayload,
+) (types.ReplayResultCollector, error) {
+	res, _, err := cr.ReplayBlockEvents(transactionEvents, blockEvent)
+	return res, err
+}
+
+// ReplayBlockEvents replays the execution of the transactions of an EVM block
 // using the provided transactionEvents and blockEvents,
-// which include all the context data for re-executing the transactions, and returns the replay result.
+// which include all the context data for re-executing the transactions, and returns
+// the replay result and the result of each transaction.
+// the replay result contains the register updates, and the result of each transaction
+// contains the execution result of each transaction, which is useful for recontstructing
+// the EVM block proposal.
 // this method can be called concurrently if underlying storage
 // tracer and block snapshot provider support concurrency.
 //
 // Warning! the list of transaction events has to be sorted based on their
 // execution, sometimes the access node might return events out of order
 // it needs to be sorted by txIndex and eventIndex respectively.
-func (cr *Replayer) ReplayBlock(
+func (cr *Replayer) ReplayBlockEvents(
 	transactionEvents []events.TransactionEventPayload,
 	blockEvent *events.BlockEventPayload,
-) (types.ReplayResultCollector, error) {
+) (types.ReplayResultCollector, []*types.Result, error) {
```

Comment:
Added the result for comparing trie updates, but the original `ReplayBlock` is unchanged, so that gateway doesn't need to update its usage.